[patch] Consolidate IBM entitlement management

durera opened the pull request:
## Entitlement
Simplify how IBM entitlement is handled during the install. A single prompt for an IBM Entitlement Key replaces individual prompts for MAS, CP4D, and SLS entitlement keys.  Only when the installer is used in a development mode will there be seperate prompts.

## Makefile
Adds a simple Makefile to simplify development

## Cloud Pak for Data
Adds basic enablement of CP4D to the one-click installer.  Although not properly integrated into the pipeline yet, the installer will prompt the user with a simple `y/N`  option to install Cloud Pak for Data.

v10.1.0 of ansible-devops also removes the need to apply the global image pull secret for CP4D, so the `cp4d_hack_worker_nodes` role has been removed from provision-roks

alequint commented on bin/templates/pipeline.yaml:
```diff
@@ -683,6 +818,8 @@ spec:
         - suite-verify
         - kafka
+        # Add this to prevent applicaton installs starting until Core FVT has completed
+        - fvt-smtp
```
Ops... will remove it

alequint commented on bin/templates/pipeline.yaml:
```diff
@@ -1039,13 +1238,20 @@ spec:
-        - app-cfg-manage
+        - suite-verify
+        # Add this to prevent applicaton installs starting until Core FVT has completed
+        - fvt-smtp
```
will remove it

durera committed:
Update Dockerfile

durera committed:
Entitlement mgmt consolidation

durera committed:
Updates

durera committed:
Support cpd_product_version

durera committed:
Create Makefile

durera committed:
Update to ansible-airgap:1.1.0 base

durera committed:
Use 10.1.0 pipeline tasks definitions

durera committed:
Remove reboot worker nodes from provision-roks

Also sets default version of OCP to 4.10 now that we've tested with MAS 8.7 & 8.8.
